### DISCUSSION TITLE
How to trigger action from release, ignoring specific tags

### DISCUSSION THREAD

**Discussion by alexfoxy** [community/community#25312] [Actions]
<p>Hi,</p>
<p>I have two actions, deploy-staging and deploy-production that I want to trigger from releases e.g. “v1.0.0-beta” (for staging) and “v1.0.0” (for production).</p>
<p>I am trying to use the tags.created with a wild card match but both actions always trigger. See below:</p>
<p>deploy-staging:</p>
<pre><code class="lang-yaml">on:
  release:
    types: created
    tags:
      - 'v*-beta*'
</code></pre>
<p>deploy-production:</p>
<pre><code class="lang-yaml">on:
  release:
    types: created
    tags:
      - '!v*-beta*'
</code></pre>
<p>Any idea how best to go about this? This should be simple to do, the docs say to use a “matching negative pattern” which I am but either I’m doing something wrong or it’s broken.</p>
<p>Thanks!<br>
Alex</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/alexfoxy">@alexfoxy</a>,</p>
<p><strong>Tag filters</strong> are not evaluated for release event, they are only available to <strong>push</strong> event and <strong>pull_request</strong> event. To view more details, you can see <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">here</a>.</p>
<p>As the workaround, you can try using the <strong>if</strong> conditional to skip the jobs in the workflow.<br>
The <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github.ref</strong></a> of the workflow run triggered by the release event is the released tag, you can use the job-level <strong>if</strong> conditional (<a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener"><strong>jobs.&lt;job_id&gt;.if</strong></a>) to skip the jobs in the workflow according to the version of the  tag.<br>
For example:</p>
<ol>
<li>deploy-staging</li>
</ol>
<pre><code class="lang-yaml">on:
  release:
    types: [created]
jobs:
  deploy_staging:
    if: ${{ contains(github.ref, '-beta') }}
    . . .
    steps:
      ---- execute steps for deploy_staging ----
</code></pre>
<ol start="2">
<li>deploy-production</li>
</ol>
<pre><code class="lang-yaml">on:
  release:
    types: [created]
jobs:
  deploy_production:
    if: ${{ !contains(github.ref, '-beta') }}
    . . .
    steps:
      ---- execute steps for deploy_production ----
</code></pre>
<p>Related docs:<br>
<a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#operators" rel="nofollow noopener"><strong>Operators</strong></a><br>
<a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions" rel="nofollow noopener"><strong>Functions</strong></a></p>
<p>Of course, if your projects really need the feature that tag filters are also available to release event, I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by alexfoxy** [comment]
<p>Thanks! This works for now :slight_smile:</p>
